File: micronaut-camunda-bpm-feature/src/test/java/info/novatec/micronaut/camunda/bpm/feature/test/ProcessEngineFactoryTest.java
package info.novatec.micronaut.camunda.bpm.feature.test;

import info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory;
import io.micronaut.test.annotation.MicronautTest;
import org.camunda.bpm.engine.*;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;

import static org.junit.jupiter.api.Assertions.*;

@MicronautTest
class ProcessEngineFactoryTest {

    @Inject
    ProcessEngine processEngine;

    @Test
    void processEngineAvailableInApplicationContext() {
        assertNotNull(processEngine);
    }

    @Test
    void testDeploymentName() {
        assertEquals(ProcessEngineFactory.MICRONAUT_AUTO_DEPLOYMENT_NAME, processEngine.getRepositoryService().createDeploymentQuery().singleResult().getName());
    }

    @Test
    void testConfigurationCustomizer() {
        assertEquals(MicronautProcessEngineConfigurationCustomizer.PROCESS_ENGINE_NAME, processEngine.getName());
    }
}

File: micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/ProcessEngineFactoryTest.kt
package info.novatec.micronaut.camunda.bpm.feature.test

import info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import org.camunda.bpm.engine.ProcessEngine
import org.camunda.bpm.engine.RepositoryService
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import javax.inject.Inject

@MicronautTest
class ProcessEngineFactoryTest {
    @Inject
    lateinit var processEngine: ProcessEngine

    @Inject
    lateinit var repositoryService: RepositoryService

    @Test
    fun `processEngine available in ApplicationContext`() {
        assertNotNull(processEngine)
    }

    @Test
    fun `auto deployment sets deployment name`() {
        assertEquals(1, repositoryService.createDeploymentQuery()
            .deploymentName(ProcessEngineFactory.MICRONAUT_AUTO_DEPLOYMENT_NAME)
            .list()
            .size)
    }

    @Test
    fun `configuration customizer`() {
        assertEquals(MicronautProcessEngineConfigurationCustomizer.PROCESS_ENGINE_NAME, processEngine.name)
    }
}